aten/src/ATen/native/miopen/Conv_miopen.cpp

```diff
@@ -351,6 +352,49 @@ struct algorithm_search<miopenConvFwdAlgorithm_t> {
         false));
     return perf_results;
   }
+
+  static miopenConvSolution_t getSolution(const ConvolutionArgs& args, bool force_default) {
+    size_t max_solution_count;
+    size_t solution_count;
+    miopenConvSolution_t solutions[AT_MIOPEN_MAX_SOLUTIONS];
+    MIOPEN_CHECK(miopenConvolutionForwardGetSolutionCount(
+        args.handle,
+        args.wdesc.desc(),
+        args.idesc.desc(),
+        args.cdesc.desc(),
+        args.odesc.desc(),
+        &max_solution_count));
+    if (max_solution_count > AT_MIOPEN_MAX_SOLUTIONS) {
+        AT_ERROR("miopenConvFwdAlgorithm_t getSolution max_solution_count > AT_MIOPEN_MAX_SOLUTIONS");
+    }
```

Comment:
@jeffdaily If there is not enough space in `solution` for all available solutions, the you can simply read as many items as you have space for:
```C++
if (max_solution_count > AT_MIOPEN_MAX_SOLUTIONS)
    max_solution_count = AT_MIOPEN_MAX_SOLUTIONS;
```
Or, even better, do not query how many solutions are available, just pass AT_MIOPEN_MAX_SOLUTIONS to `miopenConvolutionForwardGetSolution()` and it will fill the `solutions` array.

Reply:
In practice will we ever have more than the current setting of 10 for AT_MIOPEN_MAX_SOLUTIONS?  We haven't seen this check trigger an error yet.